include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -616,15 +661,20 @@ struct __parallel_radix_sort_iteration
 
         // 1. Count Phase
         sycl::event __count_event = __radix_sort_count_submit<_RadixCountKernel, __radix_bits, __is_ascending>(
-            __exec, __segments, __block_size, __radix_offset, __in_rng, __tmp_buf, __dependency_event, __proj
+            __exec, __segments, __block_size, __radix_offset, __in_rng, __tmp_buf,
+            __dependency_event, __proj
 #if _ONEDPL_COMPILE_KERNEL
             , __count_kernel
 #endif
         );
 
+        //A flag to specify "no re-order execution" if data are the same on the current sort iteration.
+        uint16_t __no_op = 0;
+        sycl::buffer<uint16_t, 1> __no_op_flag(&__no_op, 1);
```

Comment:
There is probably not that much value in optimizing that piece of code. So you can close this conversation.

I hope that the compiler is clever enough to interleave the allocations (or cache them) and copying a single element onto a device with the `count` and `scan` kernels in the current implementation. But, it can always be verified with a profiler.

But let me add some comment:
1. You can profile the code, e.g. with [onetrace](https://github.com/intel/pti-gpu/tree/master/tools/onetrace) to make sure that there are no extra allocations.  
2. I do not see any need in barriers. You only need to fill the buffer with a single work-item (there is no cooperation). You may also call `sycl::queue::fill`: that operation may be done in parallel with `count` and `scan` kernels; there is no need to copy data between host and device with that approach.

Reply:
`sycl::queue::fill` leads to additional submit... I believe it will be a reason of bigger overhead...

`Equivalent to submitting a command-group containing handler::fill(ptr, pattern, count).`